core/include/cubos/core/al/audio_device.hpp

```diff
@@ -90,14 +86,11 @@ namespace cubos::core::al
         class CUBOS_CORE_API Buffer
         {
         public:
-            virtual ~Buffer() = default;
+            ma_decoder mDecoder;
 
-            /// @brief Fills the buffer with data.
-            /// @param format Audio format of the data.
-            /// @param size Size of the buffer in bytes.
-            /// @param data Buffer data.
-            /// @param frequency Audio frequency.
-            virtual void fill(Format format, std::size_t size, const void* data, std::size_t frequency) = 0;
+            virtual ~Buffer() = default;
+            virtual size_t getLength() = 0;
+            static std::shared_ptr<Buffer> create(const void* data, size_t dataSize);
```

Comment:
This shouldn't be static, and buffers should only be created through `AudioDevice::createBuffer`, as it is the audio device which knows which implementation is 'selected'.